### PR TITLE
deploy-script: create table and map hosts IP

### PR DESCRIPTION
**Pull request by nlbxyz**

### CODE REVIEW

File: scripts/bernstein-deploy.sh [hunk 1]
```diff
@@ -41,24 +46,27 @@ $kubectl apply -f poll.deployment.yaml \
...
 echo "[$(date)] kubectl: apply traefik conf"
 $kubectl apply -f traefik.rbac.yaml \
 -f traefik.deployment.yaml \
 -f traefik.service.yaml
-# $kubectl apply -f traefik.rbac.yaml -f traefik.deployment.yaml -f traefik.service.yaml
 
-exit
+echo "[$(date)] kubectl: create postgres table"
+pg_deploy_id_pattern="kube-deploy-postgres-*-*"
+pg_deploy_id=$($kubectl get pods -o json | jq -r '.items[]|select(.metadata.name|test("$pg_deploy_id_pattern")).metadata.name')
```
Comment (Cavonstavant) [inline scripts/bernstein-deploy.sh:57]: OMFG that's so smart !!! :rocket:

File: scripts/bernstein-deploy.sh [hunk 2]
```diff
@@ -41,24 +46,27 @@ $kubectl apply -f poll.deployment.yaml \
...
-- psql -U username -P password
-# echo "CREATE TABLE votes(id text PRIMARY KEY, vote text NOT NULL);" | $kubectl exec -i <postgres-deployment-id> -c <postgres-container-id> - psql -U <username>
+echo "CREATE TABLE votes(id text PRIMARY KEY, vote text NOT NULL);" \
+| $kubectl exec -i $pg_deploy_id - psql -U $psql_username -P $psql_password
 
+echo "[$(date)] system: *.dop.io IPs mapping"
 echo "$($kubectl get nodes -o \
-jsonpath='{ $.items[*].status.addresses[?(@.type=="ExternalIP")].address }') \
-poll.dop.io result.dop.io" \
+jsonpath='{ $.items[*].status.addresses[?(@.type=="ExternalIP")].address }') poll.dop.io result.dop.io" \
```
Comment (Cavonstavant) [inline scripts/bernstein-deploy.sh:71]: jq is so powerfull damn :fire:

**Review by Cavonstavant — APPROVED** [review]
NICE !

File: scripts/bernstein-deploy.sh [hunk 3]
```diff
@@ -41,24 +46,27 @@ $kubectl apply -f poll.deployment.yaml \
...
+#echo "$pg_deploy_id"
+#echo "$psql_username"
+#echo "$psql_password"
 
-echo "CREATE TABLE votes \
-(id text PRIMARY KEY, vote text NOT NULL);" \
-| $kubectl exec -i <postgres-deployment-id> -c <postgres-container-id> \
-- psql -U username -P password
-# echo "CREATE TABLE votes(id text PRIMARY KEY, vote text NOT NULL);" | $kubectl exec -i <postgres-deployment-id> -c <postgres-container-id> - psql -U <username>
+echo "CREATE TABLE votes(id text PRIMARY KEY, vote text NOT NULL);" \
```
Comment (Cavonstavant) [inline scripts/bernstein-deploy.sh:66]: creating a table should only be done if the table does not exists:
```suggestion
echo "CREATE TABLE IF NOT EXISTS votes(id text PRIMARY KEY, vote text NOT NULL);" \
```

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
:warning:  Creating a table should only be done when it does not exists :warning:

File: scripts/bernstein-deploy.sh (hunk 3 above)
Comment (nlbxyz) [reply scripts/bernstein-deploy.sh:66]: Good idea

**Pull request merged**